scylla/src/frame/response/result.rs

```diff
@@ -608,6 +618,15 @@ fn deser_cql_value(typ: &ColumnType, buf: &mut &[u8]) -> StdResult<CQLValue, Par
             let uuid = uuid::Uuid::from_slice(buf).expect("Deserializing Uuid failed.");
             CQLValue::Uuid(uuid)
         }
+        Varint => {
+            let int_value;
+            if buf[0] == 0x00 {
+                int_value = num_bigint::BigInt::from_bytes_be(Sign::Plus, buf);
+            } else {
+                int_value = num_bigint::BigInt::from_bytes_be(Sign::Minus, buf);
+            }
+            CQLValue::Varint(int_value)
+        }
```

Comment:
This doesn't work for negative values. I ran `test_cql_types` and `-10000` inserted by our driver is read and printed as `-55536`.

There is a function already in the `num_bigint` library which handles the format for us - it's [`from_signed_bytes_be`](https://docs.rs/num-bigint/0.3.2/num_bigint/struct.BigInt.html#method.from_signed_bytes_be). All this code can be replaced with the following line:

```rust
Varint => CQLValue::Varint(num_bigint::BigInt::from_signed_bytes_be(buf)),
```

Reply:
Fixed!